assets/javascripts/discourse/initializers/extend-for-solved-button.js

```diff
@@ -196,6 +190,18 @@ export default {
           user_path: User.create({ username }).path,
         });
       }),
+      accepterHtml: computed("accepted_answer", function () {
+        const username = this.get("accepted_answer.accepter_username");
+        const name = this.get("accepted_answer.accepter_name");
+        const formattedUsername =
+          this.siteSettings.display_name_on_posts && name
+            ? name
+            : formatUsername(username);
```

Comment:
Perhaps a `displayName` logic that can be handled on the server? Much easier to test, logic is handled by the server and less data to ship to client. I don't really see a good reason for this logic to live in the client side application.

Reply:
We discussed this internally and realised that the frontend has alot of logic tied to `accepter_username` and `accepter_name` which will require significant refactoring.